cloudinit/config/cc_apt_configure.py

```diff
@@ -244,11 +244,24 @@ def apply_apt(cfg, cloud, target):
         )
     # GH: 4344 - stop gpg-agent/dirmgr daemons spawned by gpg key imports.
     # Daemons spawned by cloud-config.service on systemd v253 report (running)
-    subp.subp(
-        ["gpgconf", "--kill", "all"],
+    gpg_process_out, _err = subp.subp(
+        ["ps", "-o", "pid,args", "-C", "dirmngr", "-C", "gpg-agent"],
         target=target,
         capture=True,
+        rcs=[0, 1],
     )
+    gpg_pids = re.findall(r"(?P<pid>\d+).*homedir \/root", gpg_process_out)
```

Comment:
I think that relying on the cli args (e.g., `homedir \/root`) could cause problems in the future if they ever happen to change. What do you think about adding `ppid` to the ps options, and then having a regex like the following?
`r"(?P<pid>\d+)\s+1"`

It will capture the PID of every line that also has a PPID of 1, and since we specified `dirmngr` and `gpg-agent`, we shouldn't get any extra processes.

Reply:
Yes better, I'm using `ps -o ppid,pid` and comparing to PPID==1 now. thanks